File: Server/Server/Data/Models/Forum.cs
using System.ComponentModel.DataAnnotations;
using static Server.Data.Validations.Forum;

namespace Server.Data.Models
{
    public class Forum
    {
        public int Id { get; set; }
        public int? ParentId { get; set; }
        public string Type { get; set; } = "category";
        public string Name { get; set; }

        [MaxLength(MaxDescriptionLength)]
        public string Description { get; set; }
        public Forum Parent { get; set; }
        public List<Forum> Children { get; set; }
        public List<Topic> Topics { get; set; }
        public string UserId { get; set; }
        public User User { get; set; }
    }
}

File: ServerLaravel/app/Models/Forum.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Forum extends Model
{
    use HasFactory;
    protected $table = "forums";
    protected $fillable = [
        "parent_id",
        "type",
        "name",
        "description",
        "user_id",
        "image_url"
    ];

    public function children() {
        return $this->hasMany(Forum::class, "parent_id", "id");
    }

    public function topics() {
        return $this->hasMany(Topic::class, "forum_id", "id");
    }

    public function posts() {
        return $this->hasManyThrough(Post::class, Topic::class);
    }

    public function user() {
        return $this->hasOne(User::class, "id", "user_id");
    }

}
